pioinstaller/util.py

```
@@ -90,24 +85,45 @@ def safe_create_dir(path, raise_exception=False):
     assert src.endswith("tar.gz")
     with tarfile.open(src, mode="r:gz") as fp:
-        fp.extractall(dst)
+        if sys.version_info >= (3, 12):
+            fp.extractall(dst, filter="data")
+        else:
+            def _safe_members(tf):
+                dst_real = os.path.realpath(dst)
+                for m in tf.getmembers():
+                    target = os.path.realpath(os.path.join(dst, m.name))
+                    if not (target == dst_real or target.startswith(dst_real + os.sep)):
+                        raise tarfile.ExtractError(f"Blocked unsafe tar member: {m.name}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)